src/components/PaymentInfoLine.vue

```diff
@@ -82,12 +167,95 @@ class PaymentInfoLine extends Vue {
         default: 'normal',
     })
     public theme!: string;
+    private referenceRate: number | null = null;
+    private referenceRateUpdateTimeout: number = -1;
+    private lastTooltipToggle: number = -1;
+
+    private created() {
+        this.updateReferenceRate();
+    }
+
+    private destroyed() {
+        window.clearTimeout(this.referenceRateUpdateTimeout);
+    }
 
     public async setTime(time: number) {
         await this.$nextTick(); // let vue update in case the timer was just added
         if (!this.$refs.timer) return;
         (this.$refs.timer as Timer).synchronize(time);
     }
+
+    private get originDomain() {
+        return this.origin.split('://')[1];
+    }
+
+    private get effectiveRate() {
+        if (!this.fiatAmount) return null;
+        // Note: precision loss should be acceptable here
+        return this.fiatAmount.amount / (Number(this.cryptoAmount.amount) / (10 ** this.cryptoAmount.decimals));
+    }
+
+    private get formattedMerchantFee() {
+        if (this.merchantFee === null || this.merchantFee === undefined) return null;
+        // Note: precision loss should be acceptable here
+        const merchantFeePercent = Number(this.merchantFee) / Number(this.cryptoAmount.amount) * 100;
+        // Round to two decimals. Always ceil to avoid displaying a lower fee than charged.
+        return `${Math.ceil(merchantFeePercent * 1e2) / 1e2}%`;
```

Comment:
No need to write 100 in e-notation, it's the same length of three characters, but more confusing :laughing: 
```suggestion
        return `${Math.ceil(merchantFeePercent * 100) / 100}%`;
```

Reply:
It was actually written like this to make the intention clearer ;)